src/ray/util/util.h

```diff
@@ -35,4 +38,33 @@ inline ray::Status boost_to_ray_status(const boost::system::error_code &error) {
   }
 }
 
+template <class Shutdown>
+class InitShutdownWrapper {
```

Comment:
To me "Wrapper" is not an informative name.  How about "Raii"?  "RayInitShutdownRaii" (or RAII)
https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization